### PR TITLE
[CALCITE-4931] Upgrade SLF4J binding to Log4j2 version 2.15.0

### PR DESCRIPTION
**Pull request by zabetak**
Log4j (binding) is used for testing purposes in various modules and
for production code (shaded) in standalone-server and tck modules.

1. Replace slf4j-log4j12 dependency (using Log4j 1.x)  with
log4j-slf4j-impl (using Log4j 2.x) and take latest version
2. Use XML syntax instead of property syntax for Log4j configuration. A
Log4j configuration is hierarchical by nature so formats with natural
support for nesting (such as XML) are easier to use.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -77,3 +77,4 @@ protobuf.version=3.17.1
 scott-data-hsqldb.version=0.1
 servlet.version=4.0.1
 slf4j.version=1.7.25
+log4j2.version=2.15.0
```
Comment (julianhyde) [inline gradle.properties:80]: can you move this into alphabetical order

**Comment by zabetak** [comment]
Logs in tests seems to work fine and when I start the standalone-server I see the expected logs.

```
./gradlew build
cd standalone-server/build/libs/
java -jar avatica-standalone-server-1.20.0-SNAPSHOT-shadow.jar -u "jdbc:super://localhost"
```

**Review by julianhyde — APPROVED** [review]
+1

**Review by joshelser — APPROVED** [review]
Lgtm

File: gradle.properties (hunk 1 above)
Comment (joshelser) [reply gradle.properties:80]: fixing this and committing now.

**Pull request closed without merging**